### DISCUSSION TITLE
If Else syntax in HA-Floorplan

### DISCUSSION THREAD

**Discussion by GRJ74** [ExperienceLovelace/ha-floorplan#334] [Q&A] [answered]
I’m trying to make an if/else statement to change the background on the floorplan depending on the actual temperature compared to the setpoint. I can change the colour on if I cut the if/else part so I believe that the error is in the syntax of the if/else statement. 
 
![image](https://user-images.githubusercontent.com/123829327/233469646-730caa68-1b82-4ae5-bc69-c2fd55a0b146.png)

I’ve tried different variants with or without { } : ; % but no luck so far. 
The none working code is:
Edit:
The working code

```
     - name: diff_temp
      entities:
        - sensor.avvik_kontor
        - sensor.avvik_entre
        - sensor.avvik_stue
        - sensor.avvik_sov1
        - sensor.avvik_sov2
        - sensor.avvik_sov3
        - sensor.avvik_sov4
      state_action:
        service: floorplan.class_set
        service_data:
          class: |
            >
             var diff = entity.state;
             if (diff > 1.5) return "avvik-plus2";
             else if (diff > 0.6) return "avvik-plus1"; 
             else if (diff < -1.5) return "avvik-minus2";
             else if (diff < -0.6) return "avvik-minus1";
      tap_action: none
```
CSS-file

```
.avvik-plus2 {
  fill: #FF2A2A !important;
  fill-opacity: 0.7 !important;
}
.avvik-plus1 {
  fill: #FFAAAA !important;
  fill-opacity: 0.7 !important;
}
.avvik-minus1 {
  fill: #AACCFF !important;
  fill-opacity: 0.7 !important;
}
.avvik-minus2 {
  fill: #2A7FFF !important;
  fill-opacity: 0.7 !important;
}
```

And resulting floorplan

![image](https://user-images.githubusercontent.com/123829327/233486591-f201dae4-cdc1-48ac-86f0-c7458c68f05d.png)

**Comment by exetico** [comment, marked as answer]
You'll need to adjust the syntax a bit: See the Javascript one here:
https://www.w3schools.com/js/js_if_else.asp

I'll suggest you to check our discussions and examples? There's plenty of if-else statements, so I'm wondering if you didn't search at all 😄 . A simpel search on `else` gives plenty of results.

There's a normal JS statement here:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/68

Here's another way you can do the if-else statement:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/324

Try and give the search function another visit, if you're still missing examples.

**Reply by GRJ74** [reply]
Thank you very much!
I’ve been reading both Github and HA forum and trying a lot of them, included syntax from #324. The #68 has passed me until now and that solved the case

**Reply by exetico** [reply]
That's great to hear 😊! Fell free to share the result im the first post, by making a edit.
